File: spaced-learning-api/src/main/java/com/spacedlearning/entity/enums/SetCategory.java
package com.spacedlearning.entity.enums;

import lombok.Getter;

/**
 * Enum representing possible set categories for statistics and filtering.
 */
@Getter
public enum SetCategory {
    VOCABULARY("vocabulary"),
    GRAMMAR("grammar"),
    MIXED("mixed"),
    OTHER("other");

    private final String value;

    SetCategory(String value) {
        this.value = value;
    }

    @Override
    public String toString() {
        return value;
    }
}

File: spaced_learning_app/lib/features/learning_set/domain/entities/set_category.dart
import 'package:freezed_annotation/freezed_annotation.dart';

@JsonEnum(unknownEnumValue: SetCategory.unknown)
enum SetCategory {
  @JsonValue('vocabulary') vocabulary,
  @JsonValue('grammar') grammar,
  @JsonValue('mixed') mixed,
  @JsonValue('other') other,
  unknown,
}
